File: python/number-of-distinct-averages.py
class Solution:
    def distinctAverages(self, nums: List[int]) -> int:
        averages = set([])
        nums.sort()
        for i in range(len(nums) // 2):
            averages.add((nums[i] + nums[len(nums) - i - 1]) / 2)
        return len(averages)

File: rust/number-of-distinct-averages.rs
use std::collections::HashSet;

impl Solution {
    pub fn distinct_averages(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut averages = HashSet::new();
        for i in 0..(nums.len() / 2) {
            let average = (nums[i] as f64 + nums[nums.len() - i - 1] as f64) / 2.0;
            averages.insert(average.to_string());
        }
        return averages.len() as i32;
    }
}
